api/src/routes/donate.ts

```diff
@@ -214,3 +201,164 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
+    typescript: true
+  });
+
+  // @ts-expect-error - @fastify/csrf-protection needs to update their types
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  fastify.addHook('onRequest', fastify.csrfProtection);
+
+  fastify.post(
+    '/donate/create-stripe-payment-intent',
+    {
+      schema: schemas.createStripePaymentIntent
+    },
+    async (req, reply) => {
+      const { email, name, amount, duration } = req.body;
+
+      if (
+        !isEmail(email) ||
+        !donationSubscriptionConfig.plans[duration].includes(amount)
+      ) {
+        void reply.code(400);
+        return {
+          error: 'The donation form had invalid values for this submission.'
+        } as const;
+      }
+
+      try {
+        const stripeCustomer = await stripe.customers.create({
+          email,
+          name
+        });
+
+        const stripeSubscription = await stripe.subscriptions.create({
+          customer: stripeCustomer.id,
+          items: [
+            {
+              plan: `${donationSubscriptionConfig.duration[duration]}-donation-${amount}`
+            }
+          ],
+          payment_behavior: 'default_incomplete',
+          payment_settings: { save_default_payment_method: 'on_subscription' },
+          expand: ['latest_invoice.payment_intent']
+        });
+
+        if (
+          stripeSubscription.latest_invoice &&
+          typeof stripeSubscription.latest_invoice !== 'string' &&
+          stripeSubscription.latest_invoice.payment_intent &&
+          typeof stripeSubscription.latest_invoice.payment_intent !==
+            'string' &&
+          stripeSubscription.latest_invoice.payment_intent.client_secret !==
+            null
+        ) {
+          const clientSecret =
+            stripeSubscription.latest_invoice.payment_intent.client_secret;
+          return reply.send({
+            subscriptionId: stripeSubscription.id,
+            clientSecret
+          });
+        } else {
+          throw new Error('Stripe payment intent client secret is missing');
+        }
+      } catch (error) {
+        fastify.log.error(error);
+        void reply.code(500);
+        return reply.send({
+          error: 'Donation failed due to a server error.'
+        });
+      }
+    }
+  );
+
+  fastify.post(
+    '/donate/charge-stripe',
+    {
+      schema: schemas.chargeStripe
+    },
+    async (req, reply) => {
+      try {
+        const { email, amount, duration, subscriptionId } = req.body;
+
+        const subscription =
+          await stripe.subscriptions.retrieve(subscriptionId);
+        const isSubscriptionActive = subscription.status === 'active';
+        const productId =
+          subscription.items?.data[0]?.plan?.product?.toString();
+        const isStartedRecently = inLastFiveMinutes(
+          subscription.current_period_start
+        );
+        const isProductIdValid =
+          typeof productId == 'string' &&
+          allStripeProductIdsArray.includes(productId);
```

Comment:
```suggestion
        const isProductIdValid =
          productId && allStripeProductIdsArray.includes(productId);
```
Nit: bit cleaner, imo.